ci/test_cpp.sh

```diff
@@ -23,11 +23,15 @@ rapids-print-env
 rapids-logger "Check GPU usage"
 nvidia-smi
 
+EXITCODE=0
+trap "EXITCODE=1" ERR
+set +e
+
 rapids-logger "Begin cpp tests"
 cmake -S testing -B build
 
 cd build
 ctest --schedule-random --output-on-failure
```

Comment:
```suggestion
rapids-logger "Begin cpp tests"
cmake -S testing -B build

cd build

EXITCODE=0
trap "EXITCODE=1" ERR
set +e
ctest --schedule-random --output-on-failure
```

I think we should move this to just above the `ctest` invocation.

The other parts should fail immediately if there's an error.